client/state/data-layer/wpcom/account-recovery/index.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+} from 'state/action-types';
+
+export const requestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.undocumented().accountRecoveryReset( userData ).getResetOptions()
+		.then( options => dispatch( {
+			type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+			options,
+		} ) )
+		.catch( error => dispatch( {
+			type: ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+			error,
+		} ) )
+);
```

Comment:
In the tests I see that the request is mocked out at `/account-recovery/lookup` but this file is `/account-recovery/`. In keeping with the API file structure can we make it `state/data-layer/wpcom/account-recovery/lookup/index.js` so that we have a 1:1 mapping between WordPress.com API endpoints and the files/functions in Calypso that interact with them?

There would still be `state/data-layer/wpcom/index.js` but it would do little more than merge the `lookup` handlers into its own and pass them upstream.